src/main/java/com/google/sps/servlets/EditRecipeServlet.java

```diff
@@ -0,0 +1,120 @@
+    // String privacy = request.getParameter("privacy");
+
+    // declaring the lists
+    ArrayList<String> tags;
+    ArrayList<String> ingredients;
+    ArrayList<String> steps;
+
+    long id = Long.parseLong(idResponse);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    Key key = KeyFactory.createKey("PrivateRecipe", id);
+    try {
+      Entity recipeEntity = datastore.get(key);
+
+      // if the name input in the form is empty, the original value will be saved
+      if (name.equals("")) {
+        name = (String) recipeEntity.getProperty("name");
+      } else {
+        // formats the form response
+        name.trim();
+      }
+
+      // if the tags input in the form is empty, the original value will be saved
+      if (tagsResponse.equals("")) {
+        tags = (ArrayList<String>) recipeEntity.getProperty("tags");
+      } else {
+        // splits the String responses from the form by commas, trims each member of array, and makes the resulting arrays into Lists
+        tags = new ArrayList<String>(Arrays.asList(Arrays.stream(tagsResponse.split(",")).map(String::trim).toArray(String[]::new)));
+        // removes any empty or null members from the list
+        tags.removeAll(Arrays.asList("", null));
+      }
+
+      // if the description input in the form is empty, the original value will be saved
+      if (description.equals("")) {
+        description = (String) recipeEntity.getProperty("description");
+      } else {
+        // formats the form response
+        description.trim();
+      }
+
+      // if the ingredients input in the form is empty, the original value will be saved
+      if (ingredientsResponse.equals("")) {
+        ingredients = (ArrayList<String>) recipeEntity.getProperty("tags");
+      } else {
+        ingredients = new ArrayList<String>(Arrays.asList(Arrays.stream(ingredientsResponse.split("\n")).map(String::trim).toArray(String[]::new)));
+        ingredients.removeAll(Arrays.asList("", null, "\n", "\r\n", "\r"));
+      }
+
+      // if the ingredients input in the form is empty, the original value will be saved
+      if (stepsResponse.equals("")) {
+        steps = (ArrayList<String>) recipeEntity.getProperty("tags");
+      } else {
+        steps = new ArrayList<String>(Arrays.asList(Arrays.stream(stepsResponse.split("\n")).map(String::trim).toArray(String[]::new)));
+        steps.removeAll(Arrays.asList("", null, "\n", "\r\n", "\r"));
+      }
+
+      recipeEntity.setProperty("name", name);
+      recipeEntity.setProperty("tags", tags);
+      recipeEntity.setProperty("description", description);
+      recipeEntity.setProperty("ingredients", ingredients);
+      recipeEntity.setProperty("steps", steps);
+      recipeEntity.setProperty("imageBlobKey", (String) recipeEntity.getProperty("imageBlobKey"));
+
+      datastore.put(recipeEntity);
+
+    } catch (EntityNotFoundException e) {
+        System.out.println("entity not found exception");
```

Comment:
tbh I don't see a scenario where the key will not point to an entity bc I get the id I'm using to make the key from the datastore which goes to the json which is sent to the frontend in a hidden input the user cannot see or modify and then back to the backend
I only put the try catch because the compiler would get mad at me otherwise
there is no future implementation

Reply:
Ok, maybe a small comment here would help then documenting that this shouldn't be entered under normal circumstance based on what the user does. That way, if someone ever got the log statement they know not to look there.

Reply:
okay cool will do